### PR TITLE
Add option for using a custom salt for ourselves

### PR DESCRIPTION
**Pull request by dgarske**
# Description

Add option for using a custom salt for ourselves. 
TODO: Add test

ZD 17988

# Testing

```
./configure --enable-eccencrypt && make
```

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfcrypt/src/ecc.c [hunk 1]
```diff
@@ -13830,7 +13830,7 @@ int wc_ecc_ctx_set_peer_salt(ecEncCtx* ctx, const byte* salt)
  *
  * @param  [in, out]  ctx   ECIES context object.
  * @param  [in]       salt  Salt to use with KDF.
- * @param  [in]       len   Length of salt in bytes.
+ * @param  [in]       sz    Length of salt in bytes.
```
Comment (SparkiDev) [inline wolfcrypt/src/ecc.c:13833]: Parameter is 'len'.
Change it back or change the name of the parameter to the function.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Comment by SparkiDev** [comment]
New function looks good to me.

**Comment by dgarske** [comment]
Retest this please

**Comment by dgarske** [comment]
Retest this please

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**